### PR TITLE
Update flyway

### PR DESCRIPTION
**Pull request by twerthi**
---

# Background

Redgate has changed their method for their licensing from a `license key` to a `personal access token`.  This PR adds the new PAT method and provides a warning that the license key method has been deprecated and will be removed in future versions.

# Results

Enterprise features successfully work using the PAT method.

## Before

The only way to use Enterprise features was to use the license key, which has been deprecated by Redgate.

## After

Enterprise features will now work using either the PAT or license key methods, a warning is displayed if the license key is used letting the user know that it will be removed in future versions.

# Pre-requisites

- [X] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [X] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [X] Parameter names should not start with `$`
- [X] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [X] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [X] The best practices documented [here](https://github.com/OctopusDeploy/Library/wiki/Best-Practices) have been applied
- [ ] If a new `Category` has been created:
   - [ ] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [ ] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

Fixes # . _If there is an open issue that this PR fixes add it here, otherwise just remove this line_

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -10536,6 +10536,7 @@
         "moment": "^2.13.0",
         "node-uuid": "^1.4.7",
         "normalize.css": "^4.1.1",
+        "octopus-library": "file:",
```
Comment (harrisonmeister) [inline package-lock.json:10539]: @twerthi - is this expected?

File: step-templates/flyway-database-migrations.json [hunk 2]
```diff
@@ -21,7 +21,7 @@
...
...st call to flyway command based on OS\nif ($IsLinux)\n{\n    & bash $flywayCmd $arguments\n}\nelse\n{\n    & $flywayCmd $arguments\n}\n\n# Check exit code\nif ($lastExitCode -ne 0)\n{\n\t# Fail the step\n    Write-Error \"Execution of Flyway failed!\"\n}\n\n$currentDate = Get-Date\n$currentDateFormatted = $currentDate.ToString(\"yyyyMMdd_HHmmss\")\n\n# Check to see if the dry run variable has a value\nif (![string]::IsNullOrWhitespace($dryRunOutputFile))\n{     \n    $sqlDryRunFile = \"$($dryRunOutputFile).sql\"\n    $htmlDryRunFile = \"$($dryRunOutputFile).html\"\n    \n    if (Test-Path $sqlDryRunFile)\n    {\n    \tNew-OctopusArtifact -Path $sqlDryRunFile -Name \"$($flywayStepName)_$($flywayEnvironment)_$($currentDateFormatted)_dryRunOutput.sql\"\n    }\n    \n    if (Test-Path $htmlDryRunFile)\n    {\n    \tNew-OctopusArtifact -Path $htmlDryRunFile -Name \"$($flywayStepName)_$($flywayEnvironment)_$($currentDateFormatted)_dryRunOutput.html\"\n    }\n}\n\n$reportFile = Join-Path $(Get-Location) \"report.html\"\n    \nif (Test-Path $reportFile)\n{\n  \tNew-OctopusArtifact -Path $reportFile -Name \"$($flywayStepName)_$($flywayEnvironment)_$($currentDateFormatted)_report.html\"\n}",
```
Comment (harrisonmeister) [inline step-templates/flyway-database-migrations.json:24]: Would a `Write-Warning` be better in place of:

```ps
Write-Host "*****WARNING***** Use of the License Key has been deprecated by Redgate and will be removed in future versions, use the Personal Access Token method instead."
```

File: package-lock.json (hunk 1 above)
Comment (twerthi) [reply package-lock.json:10539]: That's a good question, I'm not sure.  I hadn't noticed it was included, I didn't run anything that would have produced it, just updated the JSON file.

File: step-templates/flyway-database-migrations.json (hunk 2 above)
Comment (harrisonmeister) [reply step-templates/flyway-database-migrations.json:24]: Cool no worries. If it keeps it consistent thats better than changing all of the other entries imo

File: package-lock.json (hunk 1 above)
Comment (harrisonmeister) [reply package-lock.json:10539]: I'm mindful of changes to the package-lock file generally. If you arent expecting it to change, I'd probably say revert it?

**Review by harrisonmeister — COMMENTED** [review]
Change itself looks good. Just the query over the package-lock.json file

File: package-lock.json (hunk 1 above)
Comment (twerthi) [reply package-lock.json]: Pulled file from main branch and overwrote PR branch.

**Pull request merged**